src/main/java/dev/blaauwendraad/masker/json/ValueMaskers.java

```diff
@@ -258,7 +278,186 @@ public static ValueMasker.AnyValueMasker withRawValueFunction(Function<String, S
+                                                    decodedBytes[decodedIndex++] = (byte) (0x80 | ((codePoint >> 12) & 0x3f));
+                                                    decodedBytes[decodedIndex++] = (byte) (0x80 | ((codePoint >> 6) & 0x3f));
+                                                    decodedBytes[decodedIndex++] = (byte) (0x80 | (codePoint & 0x3f));
+                                                }
+                                            } else {
+                                                // dealing with characters with values between 2048 and 65536 which
+                                                // equals to 2^16 or 16 bits, which is 3 bytes of data in UTF-8 encoding
+                                                decodedBytes[decodedIndex++] = (byte) (0xe0 | (unicodeHexBytesAsChar >> 12));
+                                                decodedBytes[decodedIndex++] = (byte) (0x80 | ((unicodeHexBytesAsChar >> 6) & 0x3f));
+                                                decodedBytes[decodedIndex++] = (byte) (0x80 | (unicodeHexBytesAsChar & 0x3f));
+                                            }
+                                        } catch (IllegalArgumentException e) {
+                                            throw context.invalidJson(e.getMessage(), valueStartIndex);
+                                        }
+                                    }
+                                    default -> throw context.invalidJson("Unexpected character after '\\': '%s'".formatted((char) originalByte), encodedIndex);
+                                }
+                            }
+                        }
+                        decodedValue = new String(decodedBytes, 0, decodedIndex, StandardCharsets.UTF_8);
+                    }
+                    String maskedValue = masker.apply(decodedValue);
+                    if (maskedValue == null) {
+                        maskedValue = "null";
+                    } else {
+                        StringBuilder encoded = new StringBuilder();
+                        encoded.append("\""); // opening quote of the encoded string
+                        for (int i = 0; i < maskedValue.length(); i++) {
+                            char character = maskedValue.charAt(i);
+                            // escape all characters that need to be escaped, unicode character do not have to be
+                            // transformed into \ u form
+                            switch (character) {
+                                case '\b' -> encoded.append("\\b");
+                                case '\t' -> encoded.append("\\t");
+                                case '\n' -> encoded.append("\\n");
+                                case '\f' -> encoded.append("\\f");
+                                case '\r' -> encoded.append("\\r");
+                                case '"', '\\' -> encoded.append("\\").append(character);
```

Comment:
What about `\`? Is that one of those characters that don't have to be encoded?

Reply:
`\` and `"` always must be encoded, `/` might be encoded, but we don't do that

Reply:
Cool